components/basecomponents/Exercise/ExerciseOrchestrator.tsx

```diff
@@ -60,49 +53,43 @@ const Loading = () => (
 );
 
 interface ExerciseOrchestratorProps {
-  categories?: string[];
+  categoryIds?: string[];
   quickStart?: boolean;
 }
 
 // warning: language switching triggers categories props change, probably because staticpaths/props, quickstart isn't change triggered
-export const ExerciseOrchestrator = ({ categories, quickStart = false }: ExerciseOrchestratorProps) => {
+export const ExerciseOrchestrator = ({ categoryIds, quickStart = false }: ExerciseOrchestratorProps) => {
   const lang = useLanguage();
   const init = useExerciseStore((state) => state.init);
   const cleanUp = useExerciseStore((state) => state.cleanUp);
   const setLang = useExerciseStore((state) => state.setLang);
-  const setCategories = useExerciseStore((state) => state.setCategories);
-  const getCategoryNames = useExerciseStore((state) => state.getCategoryNames);
-  const getMetacategoryNames = useExerciseStore((state) => state.getMetacategoryNames);
-  const selectedCategories = useExerciseStore((state) => state.categories);
+
   const status = useExerciseStore((state) => state.status);
   const exercise = useExerciseStore((state) => state.exercise);
   const restart = useExerciseStore((state) => state.restart);
   const counter = useExerciseStore((state) => state.counter);
   const size = useExerciseStore((state) => state.size);
-  const setSize = useExerciseStore((state) => state.setSize);
+
+  const setCategories = useExerciseStore((state) => state.setCategories);
+
   const { t } = useTranslation();
   const exerciseRef = useRef(null);
   const nextButtonRef = useRef(null);
   const exerciseStatus = exercise?.status;
-  const debug = useDebug();
-
-  const exerciseSizeList = debug ? CONFIG_BASE.debugSizeList : CONFIG_BASE.sizeList;
 
   useEffect(() => {
     setLang(lang);
   }, [setLang, lang]);
 
-  useEffect(() => {
-    if (categories === undefined || categories.length === 0) return;
-    setCategories(categories);
-  }, [setCategories, categories]);
-
   useEffect(() => {
     init(quickStart === true);
+    if (quickStart === true && categoryIds?.length) {
```

Comment:
We have to deal categories and init in separate useEffect because language switching triggers categories (newly categoryIds) prop change a and then init() is called also. This happens only for Exercise started from dictionary because it has some categoryIds passed. In general, I think it's not good to merge useEffects. I read that somewhere in docs.